rts/lra/coordinator/src/main/java/io/narayana/lra/coordinator/internal/LRARecoveryModule.java

```diff
@@ -130,6 +135,35 @@ private void doRecoverTransaction(Uid recoverUid) {
         }
     }
 
+    private void moveEntryToFailedLRAPath (final Uid failedUid) throws ObjectStoreException {
+        String failedLRAType = LongRunningAction.getFailedLraType();
+        InputObjectState inputState = _recoveryStore.read_committed(failedUid, _transactionType);
+        InputObjectState failedLRAUidState = _recoveryStore.read_committed(failedUid, failedLRAType);
+        boolean result = false;
+
+        if (inputState != null) {
+            if (failedLRAUidState != null) {
+                // Record already exists in failedLRARecord_PATH, hence removing it from the LRA_PATH
+                _recoveryStore.remove_committed(failedUid, _transactionType);
+                return;
+            }
+
+            boolean moved = _recoveryStore.write_committed(failedUid, failedLRAType, new OutputObjectState(inputState));
+
+            if (!moved) {
+                if (LRALogger.logger.isDebugEnabled()) {
+                    LRALogger.logger.debugf("Unable to move failed lra record (Uid: %s) to new location type: %s", failedUid, failedLRAType);
+                }
+            }
+            else {
+                result = _recoveryStore.remove_committed(failedUid, _transactionType);
+                if (result) {
+                    LRALogger.logger.infof("Failed lra record (Uid: %s) moved to new location type: %s", failedUid, failedLRAType);
```

Comment:
The `info` logging is just to inform that the Failed LRA record is moved to separate location.
For the case in which the failedLRA record is failed to get deleted from LRA_PATH, this would be retried in the next attempt when the recovery manager will rerun for the next iteration, and this condition ( https://github.com/jbosstm/narayana/pull/1773/files/a7cdb4f993598108c9c0a974ca70a3a5e98f80d4#diff-62917e7ece8109d31366b0cd7f3a4c0b80cc996e85374f99ede7563c453db778R145 ) will remove it and will also inform the same by the logs.

`Failed` records can only be fetched from the new path i.e. `/StateManager/BasicAction/LongRunningAction/failed`, so there will not be any duplicate record.

Reply:
`The info logging is just to inform that the Failed LRA record is moved to separate location.`) I see, I got it. Thanks.

`For the case in which the failedLRA record`) Good with the recovery manager and retries. For the second part I just wonder if there could be some race condition in fetching the records and using them. E.g. by LRAService#getFailedLRAs returning the failed ones and with the recovery manager using the duplicate. Or in case of learning the status of the LRA. If that's covered by you then ok, it's fine.

Reply:
I will synchronize the loose ends.